In [74]:
import pandas as pd
import numpy as np
man_pl = pd.read_csv("man_pl(in).csv")
man_gk = pd.read_csv("man_gk.csv")
manu = pd.read_csv("man_u.csv")
tot_gk = pd.read_csv("tot_goalkeep(Sheet1).csv")
tot_pl = pd.read_csv("tot_players(Sheet1).csv")
tot = pd.read_csv("tott(Sheet2).csv")
overall =pd.read_csv("overall(Sheet1).csv")



In [25]:
print(tot.columns.tolist())


['Opponent', 'Leg', 'Tot_Possession (%)', 'OPP_Possession (%)', 'Tot_Passing Acc (%)', 'Tot_Passes Completed', 'Tot_Passes Attempted', 'OPP_Passing Acc (%)', 'OPP_Passes Completed', 'OPP_Passes Attempted', 'Tot_Shots', 'Tot_Shots on Target', 'OPP_Shots', 'OPP_Shots on Target', 'Tot_Saves (%)', 'Tot_Saves Made', 'Tot_Saves Faced', 'OPP_Saves (%)', 'OPP_Saves Made', 'OPP_Saves Faced', 'Result', 'G_OP', 'G_TOT', 'xG', 'xGA', 'Possesion']


In [42]:
#MANCHESTER UNITED IN THE LAST 6 GAMES

mavg = manu[['Possesion','MU_Shots','MU_Shots on Target','MU_Saves (%)','xG','xGA']].mean()
print(mavg)

Possesion             52.333333
MU_Shots              15.500000
MU_Shots on Target     5.833333
MU_Saves (%)          70.333333
xG                     2.433333
xGA                    1.433333
dtype: float64


In [43]:
#TOTTENHAM HOTSPUR IN THE LAST 6 GAMES

totavg = tot[['Possesion','Tot_Shots','Tot_Shots on Target','Tot_Saves (%)','xG','xGA']].mean()
print(totavg)

Possesion              48.333333
Tot_Shots              12.666667
Tot_Shots on Target     3.666667
Tot_Saves (%)          66.666667
xG                      1.450000
xGA                     0.800000
dtype: float64


In [34]:
man_pl.columns.tolist()

['Name',
 'POS',
 'Age',
 'HT',
 'WT',
 'NAT',
 'APP',
 'SUB',
 'G',
 'A',
 'SH',
 'ST',
 'FC',
 'FA']

In [44]:
man_pl.drop(columns =['POS','Age','HT','WT','NAT','SUB','FC','FA'],inplace = True)

In [45]:
man_pl.head()

Name  APP  G  A  SH  ST
0    Victor Lindelöf   13  0  0   0   0
1  Noussair Mazraoui   34  0  1  12   2
2   Matthijs de Ligt   29  0  0  16   7
3      Harry Maguire   24  2  1  12   5
4  Lisandro Martínez   20  0  1  16   6

In [46]:
tot_pl.drop(columns =['POS','Age','HT','WT','NAT','SUB','FC','FA'],inplace = True)

In [47]:
tot_pl.head()

Name  APP  G  A  SH  ST
0  Sergio Reguilón    3  0  0   1   0
1      Kevin Danso    7  0  0   6   2
2    Radu Dragusin   16  0  0   5   1
3   Destiny Udogie   24  0  1   6   2
4  Cristian Romero   18  0  0  16   7

MANCHESTER UNITED

In [52]:
top_s = man_pl[man_pl['G']>0].sort_values(by='G',ascending=False).reset_index(drop=True)
print(top_s[['Name','G']])

                  Name  G
0      Bruno Fernandes  7
1       Rasmus Højlund  6
2          Diogo Dalot  3
3        Harry Maguire  2
4          Mason Mount  2
5    Christian Eriksen  2
6             Casemiro  2
7        Kobbie Mainoo  2
8       Joshua Zirkzee  2
9                 Amad  2
10           Leny Yoro  1
11       Manuel Ugarte  1
12  Alejandro Garnacho  1


In [56]:

top_s['acc%'] = (top_s['ST']/top_s['SH'])*100
top_s[['Name','SH','ST','acc%']].sort_values(by='acc%',ascending=False)

Name  SH  ST       acc%
3        Harry Maguire  12   5  41.666667
4          Mason Mount  12   5  41.666667
1       Rasmus Højlund  27  11  40.740741
8       Joshua Zirkzee  27  11  40.740741
9                 Amad  40  14  35.000000
12  Alejandro Garnacho  83  28  33.734940
0      Bruno Fernandes  84  27  32.142857
6             Casemiro  30   8  26.666667
11       Manuel Ugarte  18   4  22.222222
2          Diogo Dalot  23   5  21.739130
10           Leny Yoro   5   1  20.000000
7        Kobbie Mainoo  11   2  18.181818
5    Christian Eriksen  12   0   0.000000

In [59]:
man_gk.head(2)

Name  Appearances  Saves  Goals Against
0  Altay Bayindir          2.0    9.0            1.0
1     André Onana         12.0   32.0           17.0

In [65]:
man_gk['s_p'] = (man_gk['Saves']/(man_gk['Saves']+man_gk['Goals Against']))*100 #save percentage
man_gk['con'] = man_gk['Goals Against']/man_gk['Appearances'] #goals conceded per match
man_gk['saved']= man_gk['Saves']/man_gk['Appearances'] #goals saved per match


man_gk[['Name','s_p','con','saved']].round(2).head(2)

Name    s_p   con  saved
0  Altay Bayindir  90.00  0.50   4.50
1     André Onana  65.31  1.42   2.67

TOTTENHAM


In [66]:
top_st = tot_pl[tot_pl['G']>0].sort_values(by='G',ascending=False).reset_index(drop=True)
print(top_st[['Name','G']])

                Name  G
0    Dominic Solanke  5
1    Brennan Johnson  4
2     James Maddison  3
3      Son Heung-Min  3
4        Pedro Porro  2
5    Pape Matar Sarr  2
6   Dejan Kulusevski  1
7        Richarlison  1
8         Mathys Tel  1
9        Timo Werner  1
10       Mikey Moore  1


In [67]:
top_st['acc%'] = (top_st['ST']/top_st['SH'])*100
top_st[['Name','SH','ST','acc%']].sort_values(by='acc%',ascending=False)

Name  SH  ST       acc%
3      Son Heung-Min  53  25  47.169811
2     James Maddison  38  17  44.736842
5    Pape Matar Sarr  31  13  41.935484
0    Dominic Solanke  60  25  41.666667
8         Mathys Tel  20   8  40.000000
7        Richarlison  18   7  38.888889
1    Brennan Johnson  52  20  38.461538
9        Timo Werner   8   3  37.500000
6   Dejan Kulusevski  46  17  36.956522
4        Pedro Porro  37   8  21.621622
10       Mikey Moore   3   0   0.000000

In [68]:
tot_gk.head()

Name  APP  SV  GA
0  Guglielmo Vicario1   23  66  31
1    Fraser Forster20    7  29  15
2    Antonin Kinsky31    4  12   5

In [71]:
tot_gk['s_p'] = (tot_gk['SV']/(tot_gk['SV']+tot_gk['GA']))*100 #save percentage
tot_gk['con'] = tot_gk['GA']/tot_gk['APP'] #goals conceded per match
tot_gk['saved']= tot_gk['SV']/tot_gk['APP'] #goals saved per match


tot_gk[['Name','APP','s_p','con','saved']].round(2)

Name  APP    s_p   con  saved
0  Guglielmo Vicario1   23  68.04  1.35   2.87
1    Fraser Forster20    7  65.91  2.14   4.14
2    Antonin Kinsky31    4  70.59  1.25   3.00

In [83]:
#MANCHESTER UNITED & TOTTENHAM EUROPA LEAGUE OVERALL STATS

print(overall[['Metric','Man U','Tottenham']].to_string(index=False))

           Metric  Man U  Tottenham
   Matches played     14         14
              Won      9          9
            Draws      5          3
             Lost      0          2
        Tot goals     35         27
       Pen scored      4          4
   Attempt on tar     87         74
    Attem blocked     68         50
          Attacks    577        486
          Assists     26         20
    Corners taken     80         92
         Offsides     20         20
  Balls recovered    578        592
          Tackles    193        158
Clarence attempts    245        256
       Goal saves     41         42
   Goals conceded     18         13
     Clean sheets      3          5


𝑴𝒀 𝑷𝑹𝑬𝑫𝑰𝑪𝑻𝑰𝑶𝑵𝑺 

OVERALL
Comparing the overall statistics I can see that Man United have not lost a single game, whereas Tottenham have lost two games, so this gives United a slight edge.
Considering Total goals, attempts on target, attacks and assists. I can conclude that man United have better performance compared to Tottenham.
Based on Balls recovered,Tackles,Clarence attempts,Goal saves,Goals conceded and Clean sheets: Man united seems to rely more on aggressive defending (more tackles), but Tottenham appears to be more effective defensively overall, conceding fewer goals and keeping more clean sheets

PLAYERS

 * For Man United; the top 3 scorers are: 

 Bruno Fernandes  -- 7

 Rasmus Højlund -- 6

 Diogo Dalot -- 3

------------------------------------------------------------------------------

and each of the 3 players have a goal shooting accuracy of 
 
 Bruno Fernandes -- 32.14%

 Rasmus Højlund -- 40.74%

 Diogo Dalot -- 21.73%

------------------------------------------------------------------------------

top 4 players that have the highest shooting accuracy are:
    
 Harry Maguire--> shots =12, shots on target-->	5, shooting accuracy --> 41.666667

 Mason Mount --> shots = 12,  shots on target--> 5, shooting accuracy --> 41.666667

 Rasmus Højlund --> shots =	27, shots on target--> 11, shooting accuracy --> 40.740741

 Joshua Zirkzee --> shots =	27, hots on target--> 11, shooting accuracy --> 40.740741

 out of the above 4 only Rasmus Højlund managed to make it in the top 3 scorers list.


------------------------------------------------------------------------------

 * For Tottenham;the top 3 scorers are:

 Dominic Solanke -- 5

 Brennan Johnson --  4

 James Maddison -- 3

 ------------------------------------------------------------------------------

and each of the 3 players have a goal shooting accuracy of 

 Dominic Solanke -- 41.66%

 Brennan Johnson -- 38.46%

 James Maddison --  44.73%

 ------------------------------------------------------------------------------

top 4 players that have the highest shooting accuracy are:

Son Heung-Min --> shots = 53, shots on target --> 25, shooting accuracy --> 47.169811

James Maddison --> shots = 38, shots on target --> 17, shooting accuracy --> 44.736842

Pape Matar Sarr --> shots = 31, shots on target --> 13, shooting accuracy --> 41.935484

Dominic Solanke --> shots = 60, shots on target --> 25, shooting accuracy --> 41.666667


out of the above 4, 2 players Dominic Solanke and James Maddison managed to make it in the top 3 scorers list.

------------------------------------------------------------------------------

Looking at the data united and spurs both have a chance of scoring but spurs most likely will manage to score atleast 1 goal more than united and winning the game because👇

 
𝟏.𝐑𝐚𝐬𝐦𝐮𝐬 𝐇ø𝐣𝐥𝐮𝐧𝐝 𝐢𝐬 𝐛𝐨𝐭𝐡 𝐚 𝐭𝐨𝐩 𝐬𝐜𝐨𝐫𝐞𝐫 𝐚𝐧𝐝 𝐨𝐧𝐞 𝐨𝐟 𝐭𝐡𝐞 𝐦𝐨𝐬𝐭 𝐚𝐜𝐜𝐮𝐫𝐚𝐭𝐞 𝐬𝐡𝐨𝐨𝐭𝐞𝐫𝐬(united).

𝟐.𝐁𝐫𝐮𝐧𝐨 𝐅𝐞𝐫𝐧𝐚𝐧𝐝𝐞𝐬 𝐬𝐜𝐨𝐫𝐞𝐬 𝐭𝐡𝐞 𝐦𝐨𝐬𝐭 𝐠𝐨𝐚𝐥𝐬 𝐛𝐮𝐭 𝐡𝐚𝐬 𝐥𝐨𝐰𝐞𝐫 𝐬𝐡𝐨𝐨𝐭𝐢𝐧𝐠 𝐚𝐜𝐜𝐮𝐫𝐚𝐜𝐲 𝐭𝐡𝐚𝐧 𝐨𝐭𝐡𝐞𝐫𝐬(united).

𝟑.𝐉𝐚𝐦𝐞𝐬 𝐌𝐚𝐝𝐝𝐢𝐬𝐨𝐧 𝐢𝐬 𝐛𝐨𝐭𝐡 𝐚 𝐭𝐨𝐩 𝐬𝐜𝐨𝐫𝐞𝐫 𝐚𝐧𝐝 𝐡𝐚𝐬 𝐯𝐞𝐫𝐲 𝐡𝐢𝐠𝐡 𝐬𝐡𝐨𝐨𝐭𝐢𝐧𝐠 𝐚𝐜𝐜𝐮𝐫𝐚𝐜𝐲(spurs).

𝟒.𝐃𝐨𝐦𝐢𝐧𝐢𝐜 𝐒𝐨𝐥𝐚𝐧𝐤𝐞 𝐢𝐬 𝐚 𝐤𝐞𝐲 𝐟𝐢𝐧𝐢𝐬𝐡𝐞𝐫, 𝐰𝐢𝐭𝐡 𝐠𝐨𝐨𝐝 𝐚𝐜𝐜𝐮𝐫𝐚𝐜𝐲 𝐚𝐧𝐝 𝐚 𝐡𝐢𝐠𝐡 𝐧𝐮𝐦𝐛𝐞𝐫 𝐨𝐟 𝐬𝐡𝐨𝐭𝐬(spurs).

 ---------------

𝑹𝒂𝒔𝒎𝒖𝒔 𝑯ø𝒋𝒍𝒖𝒏𝒅,𝑩𝒓𝒖𝒏𝒐 𝑭𝒆𝒓𝒏𝒂𝒏𝒅𝒆𝒔 𝒍𝒊𝒌𝒆𝒍𝒚 𝒕𝒐 𝒔𝒄𝒐𝒓𝒆 𝒊𝒏 𝒕𝒉𝒆 𝒇𝒊𝒏𝒂𝒍𝒔 𝒇𝒐𝒓 𝑼𝒏𝒊𝒕𝒆𝒅 ,𝑫𝒐𝒎𝒊𝒏𝒊𝒄 𝑺𝒐𝒍𝒂𝒏𝒌𝒆 𝒎𝒐𝒔𝒕 𝒍𝒊𝒌𝒆𝒍𝒚 𝒕𝒐 𝒔𝒄𝒐𝒓𝒆 𝒇𝒐𝒓 𝑺𝒑𝒖𝒓𝒔.
𝑨 𝒉𝒊𝒈𝒉 𝒕𝒉𝒓𝒆𝒂𝒕 𝒇𝒐𝒓 𝑼𝒏𝒊𝒕𝒆𝒅 -𝑺𝒐𝒏 𝒂𝒏𝒅 𝑴𝒂𝒅𝒅𝒊𝒔𝒐𝒏- 𝒑𝒍𝒂𝒚𝒆𝒓𝒔 𝒘𝒊𝒕𝒉 𝒇𝒆𝒘𝒆𝒓 𝒔𝒉𝒐𝒕𝒔 𝒃𝒖𝒕 𝒗𝒆𝒓𝒚 𝒉𝒊𝒈𝒉 𝒂𝒄𝒄𝒖𝒓𝒂𝒄𝒚 𝒄𝒂𝒏 𝒃𝒆 𝒅𝒆𝒂𝒅𝒍𝒚 𝒆𝒗𝒆𝒏 𝒊𝒇 𝒕𝒉𝒆𝒚 𝒅𝒐𝒏’𝒕 𝒕𝒐𝒑 𝒕𝒉𝒆 𝒔𝒉𝒐𝒕 𝒄𝒉𝒂𝒓𝒕𝒔.
𝑱𝒂𝒎𝒆𝒔 𝑴𝒂𝒅𝒅𝒊𝒔𝒐𝒏 𝒘𝒊𝒍𝒍 𝒃𝒆 𝒎𝒊𝒔𝒔𝒊𝒏𝒈 𝒕𝒉𝒆 𝒇𝒊𝒏𝒂𝒍𝒔 𝒅𝒖𝒆 𝒕𝒐 𝒍𝒊𝒈𝒂𝒎𝒆𝒏𝒕 𝒊𝒏𝒋𝒖𝒓𝒚 𝒊𝒏 𝒕𝒉𝒆 𝒑𝒓𝒆𝒗𝒊𝒐𝒖𝒔 𝒎𝒂𝒕𝒄𝒉 - 𝒈𝒊𝒗𝒊𝒏𝒈 𝒖𝒏𝒊𝒕𝒆𝒅 𝒂 𝒔𝒍𝒊𝒈𝒉𝒕 𝒂𝒅𝒗𝒂𝒏𝒕𝒂𝒈𝒆.

GOALKEEPING

united:
André Onana	
Saving percentage = 65.31	

conced/match = 1.42	

saved/match = 2.67
 
 ------------------------

spurs:
Guglielmo Vicario1	
Saving percentage = 68.04	

concede/match =1.35

saved/match = 2.87

-----------------

Spurs have a chance of saving more goals attempted compared to united looking at the above data
Hence I can say spurs have a slight advantage over united and can be potential winners

Player and goalkeeping statistics show tottenham winning the match by scoring more and saving attempts made by united
overall given united's unbeaten streak , overall goals and shots united have a slightly higher chance of winning than spurs

overall one has better attack and the other has better defence but going with the overall statistics ; --- *𝐓𝐨𝐭𝐭𝐞𝐧𝐡𝐚𝐦 𝐢𝐬 𝐥𝐢𝐤𝐞𝐥𝐲 𝐭𝐨 𝐰𝐢𝐧 𝐭𝐡𝐞 𝐄𝐮𝐫𝐨𝐩𝐚 𝐥𝐞𝐚𝐠𝐮𝐞 𝟐𝟎𝟐𝟓* ---